docs/configuration.md

```diff
@@ -0,0 +1,89 @@
+it (as in the example file above).
+
+Your repo will require access to [release credentials](credentials.md) to
+[pass on those secrets](https://github.com/guardian/etag-caching/blob/9935da29e76b8b89759bcfe967cc7c1c02aa1814/.github/workflows/release.yml#L11-L13)
+to the workflow.
+
+## `sbt`
+
+### Recommended `sbt` plugins
+
+[Example `project/plugins.sbt`](https://github.com/guardian/etag-caching/blob/main/project/plugins.sbt)
+
+* [`sbt-version-policy`](https://github.com/scalacenter/sbt-version-policy) - to supply automatic compatibility-based
+  version-numbering that follows [recommended Scala semver rules](https://www.scala-lang.org/blog/2021/02/16/preventing-version-conflicts-with-versionscheme.html#early-semver-and-sbt-version-policy)
+  on your library, as well as automatically setting the [`versionScheme`](https://www.scala-sbt.org/1.x/docs/Publishing.html#Version+scheme)
+  of your project to `"early-semver"` (essential).
+* [`sbt-release`](https://github.com/sbt/sbt-release)
+* [`sbt-sonatype`](https://github.com/xerial/sbt-sonatype) - you currently need to have this in your project's
+  `plugins.sbt`, as the [`🎊 Create artifacts`](https://github.com/guardian/gha-scala-library-release-workflow/blob/7d278d4d44e30b4b4c0f6791053bdeb40b8159cb/.github/workflows/reusable-release.yml#L141-L158)
+  job expects it, but in the future, the workflow will probably supply this automatically.
+
+### Recommended `sbt` settings
+
+[Example `build.sbt`](https://github.com/guardian/etag-caching/blob/main/build.sbt)
+
+* `scalacOptions` should include `-release:11` (also known as `-java-output-version`
+  [in Scala 3](https://www.scala-lang.org/blog/2022/04/12/scala-3.1.2-released.html#changes-to-other-compatibility-flags)) -
+  the workflow will always compile with one of the most recent LTS releases of Java [supported by Scala](https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html),
+  but the generated class files will be compatible with whichever version of Java you target.
+* `releaseVersion := fromAggregatedAssessedCompatibilityWithLatestRelease().value` - to activate the
+  automatic compatibility-based version-numbering provided by the `sbt-version-policy` plugin.
+* `publish / skip := true` (rather than other legacy hacks like `publishArtifact := false`) for
+  sbt modules that don't generate artifacts (often, the 'root' project in a multi-project build). This
+  setting is respected by `sbt-version-policy` - it won't attempt to calculate compatibility on a module
+  that doesn't publish artifacts.
+* In `releaseProcess`, you'll want _fewer_ steps than
+  [the old list specified by `sbt-sonatype`](https://github.com/xerial/sbt-sonatype?tab=readme-ov-file#using-with-sbt-release-plugin),
+  now just:
+  `checkSnapshotDependencies, inquireVersions, runClean, runTest, setReleaseVersion, commitReleaseVersion, tagRelease, setNextVersion, commitNextVersion`
```

Comment:
The `commitReleaseVersion` step requires branch protection to be dropped (although not sure if that is only an intermediate requirement while a different solution is being developed?). That may be obvious enough not to merit a comment about that here though.

Reply:
Good point, added in the new 'Repo settings' section in 99340564e83d511aa77134f894eed0588a2b5acf